WcaOnRails/app/controllers/database_controller.rb

```diff
@@ -4,38 +4,41 @@ class DatabaseController < ApplicationController
   RESULTS_README_TEMPLATE = 'database/public_results_readme'
 
   def results_export
-    @sql_filename, @sql_filesize = _link_info DbDumpHelper::RESULTS_EXPORT_SQL_PERMALINK
-    @tsv_filename, @tsv_filesize = _link_info DbDumpHelper::RESULTS_EXPORT_TSV_PERMALINK
+    @sql_path, @sql_filesize = get_current_export("sql")
+    @tsv_path, @tsv_filesize = get_current_export("tsv")
+    @sql_filename = File.basename(@sql_path)
+    @tsv_filename = File.basename(@tsv_path)
 
-    @sql_rel_path = DatabaseController.rel_download_path DbDumpHelper::RESULTS_EXPORT_FOLDER, @sql_filename
-    @tsv_rel_path = DatabaseController.rel_download_path DbDumpHelper::RESULTS_EXPORT_FOLDER, @tsv_filename
-
-    @sql_perma_path = DatabaseController.rel_download_path DbDumpHelper::RESULTS_EXPORT_FOLDER, DbDumpHelper::RESULTS_EXPORT_SQL_PERMALINK
-    @tsv_perma_path = DatabaseController.rel_download_path DbDumpHelper::RESULTS_EXPORT_FOLDER, DbDumpHelper::RESULTS_EXPORT_TSV_PERMALINK
+    @sql_perma_path = "#{EnvConfig.ROOT_URL}/export/results/#{DbDumpHelper::RESULTS_EXPORT_SQL_PERMALINK}"
+    @tsv_perma_path = "#{EnvConfig.ROOT_URL}/export/results/#{DbDumpHelper::RESULTS_EXPORT_TSV_PERMALINK}"
   end
 
-  def _link_info(permalink)
-    full_permalink = DbDumpHelper::RESULTS_EXPORT_FOLDER.join(permalink)
-
-    actual_filename = File.basename File.readlink(full_permalink)
-    # Manually resolve the relative link to be independent of runtime environments
-    actual_file = DbDumpHelper::RESULTS_EXPORT_FOLDER.join(actual_filename)
-
-    filesize_bytes = File.size? actual_file
-
-    [actual_filename, filesize_bytes]
+  def sql_permalink
+    url, _ = get_current_export("sql")
+    redirect_to url, status: 301, allow_other_host: true
   end
 
-  def developer_export
-    @rel_download_path = DatabaseController.rel_download_path DbDumpHelper::DEVELOPER_EXPORT_FOLDER, DbDumpHelper::DEVELOPER_EXPORT_SQL_PERMALINK
+  def tsv_permalink
+    url, _ = get_current_export("sql")
+    redirect_to url, status: 301, allow_other_host: true
   end
 
-  def self.rel_download_path(base_folder, file_name)
-    file_path = base_folder.join file_name
-    relative_path = file_path.relative_path_from DbDumpHelper::EXPORT_PUBLIC_FOLDER.parent
+  def get_current_export(type)
```

Comment:
It's not clear which export this is referring to -- above it seems like `type` is getting values "sql" and "tsv", whereas I would expect "type of export" to be `developer` or `public_results`.

Reply:
I'll change it to file_type

Reply:
In that case: What is `current` referring to?
As far as I can tell, this controller is for both developer and public_results. Which one is the "current" one w.r.t. to this method?

Reply:
this is just for the developer_export. There is no need to get the 'current export' for the developer export, because we always overwrite that one

Reply:
In that case, please make it clear by giving the method a name like `def current_results_export(file_type)`